src/app/components/category-products/category-products.component.ts

```diff
@@ -1,11 +1,105 @@
-import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, combineLatest, of } from 'rxjs';
+import {  map, shareReplay, startWith, switchMap } from 'rxjs/operators';
+import { CategoryModel } from '../../models/category.model';
+import { ProductModel } from '../../models/product.model';
+import { CategoriesService } from '../../services/categories.service';
+import { ProductsService } from '../../services/products.service';
 
 @Component({
   selector: 'app-category-products',
   styleUrls: ['./category-products.component.scss'],
   templateUrl: './category-products.component.html',
   encapsulation: ViewEncapsulation.Emulated,
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CategoryProductsComponent {
+  readonly selectedSortingOption: FormControl = new FormControl(
+    'Featured'
+  );
+  readonly sortingOptions$: Observable<string[]> = of([
```

Comment:
You can work out with query model here,  something that will tell you: which property to sort and the direction of the sort